src/web/cache.rs

```diff
@@ -112,6 +117,40 @@ impl AfterMiddleware for CacheMiddleware {
     }
 }
 
+pub(crate) async fn cache_middleware<B>(req: AxumHttpRequest<B>, next: Next<B>) -> AxumResponse {
+    let config = req
+        .extensions()
+        .get::<Arc<Config>>()
+        .cloned()
+        .expect("missing config extension in request");
+
+    let mut response = next.run(req).await;
+
+    let cache = response
+        .extensions()
+        .get::<CachePolicy>()
+        .unwrap_or(&CachePolicy::NoCaching);
+
+    if cfg!(test) {
+        assert!(
+            !response.headers().contains_key(CACHE_CONTROL),
+            "handlers should never set their own caching headers and only use CachePolicy to control caching."
+        );
+    }
+
+    let directives = cache.render(&config);
+    if !directives.is_empty() {
+        response.headers_mut().insert(
+            CACHE_CONTROL,
+            CacheControl(directives)
+                .to_string()
+                .parse()
+                .expect("cache-control header could not be parsed"),
```

Comment:
`CacheControl` is coming from hyper, and I don't see how we could change this. 

We could migrate this to our own logic, but I would prefer doing this when iron is gone.

Reply:
It's from this crate, not hyper: https://github.com/rust-lang/docs.rs/blob/04677b19069c0d50bc58a026cf622f68e0b1c056/src/web/cache.rs#L16-L17

I would expect you to be able to change this function to return HeaderName / HeaderValue: https://github.com/rust-lang/docs.rs/blob/04677b19069c0d50bc58a026cf622f68e0b1c056/src/web/cache.rs#L45-L81

That said, I'm ok with delaying this to a later PR if it seems like too much work.

Reply:
Ah, you're talking about `CachePolicy`, not `CacheControl`. That is what confused me. 

The cache-directive rendering is suboptimal in performance anyways since it's constructed & rendered on each request dynamically. The time to parse the string into a `HeaderValue` is negiglible to that IMO. 

Since this `CacheControl` / `CacheDirective` approach is using old hyper objects that don't exist any more I would rather propose I rewrite this into using static values when hyper is gone, since the possible cache-control header values are mostly static, or depending on config.

Reply:
sounds good :)

(I assume you mean when *iron* is gone)

Reply:
actually I meant the old hyper version where these classes are coming from, but that comes with iron, true :)